tests/test_client/tests.py

```diff
@@ -284,6 +284,18 @@ def test_follow_307_and_308_redirect(self):
                 self.assertEqual(response.request['PATH_INFO'], '/post_view/')
                 self.assertEqual(response.request['REQUEST_METHOD'], method.upper())
 
+    def test_follow_307_and_308_preserves_query_string(self):
+        methods = ('get', 'post', 'head', 'options', 'put', 'patch', 'delete', 'trace')
```

Comment:
`GET` works without this patch. Extra tests are fine but should be moved to a separate commit.